### PR TITLE
Feature/convinience functions

### PR DESCRIPTION
**Pull request by saschadoemer**

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -30,11 +30,25 @@ Feel free to get in touch by visiting our https://www.lmis.de[website] or contac
...
+
+The `api` module contains interfaces, common helper classes and DTOs as well as exceptions.
+
+=== `impl` Module
+
+The `impl` module contains the implementation of the given API defined by the `api` module.
+
+=== `convenience` Module
+
+The `convenience` module contains common implementations for different problems which are not located within the API. Those could be capability setting, MQTT client creation or other common problems that are normally not part of the API.
```
Comment (Frank-Wiebeler) [inline README.adoc]: There is no convinience command or class handling the capabilities

File: agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  /**
+   * Create default MQTT connect options using the given onboarding response. The MQTT options
+   * contain a socket factory which uses the personal certificate of the endpoint.
+   *
+   * @param onboardingResponse -
+   * @return -
+   * @throws Exception -
+   */
+  public MqttConnectOptions createMqttConnectOptions(OnboardingResponse onboardingResponse)
+      throws Exception {
```
Comment (Frank-Wiebeler) [inline agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java]: This should be a dedicated business exception to ensure, that the exception already transports a message.

File: agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+   * Creating the socket factory for the MQTT options.
+   *
+   * @param certificate Certificate of the endpoint.
+   * @param password The password for the certificate.
+   * @param certificationType The type of certificate.
+   * @return -
+   * @throws Exception -
+   */
+  private SocketFactory getSocketFactory(
+      String certificate, String password, CertificationType certificationType) throws Exception {
```
Comment (Frank-Wiebeler) [inline agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java]: This should be a dedicated business exception to ensure, that the exception already transports a message.

Comment (saschadoemer) [reply agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java]: Well played, well played.

File: agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java (hunk 2 above)
Comment (saschadoemer) [reply agrirouter-api-java-convenience/src/main/java/com/dke/data/agrirouter/convenience/mqtt/client/MqttOptionService.java]: Well played, well played.

**Comment by saschadoemer** [comment]
`MessageConfirmationFunctionsService` - I'll do so, but in another branch.

**Pull request merged**